arcgisUM.py

```diff
@@ -88,39 +88,84 @@ def getArcGISGroupByTitle(arcGISAdmin, title):
     return None
 
 
-def addCanvasUsersToGroup(instructorLog, group, courseUsers):
-    """Add new users to the ArcGIS group.  """
-    groupNameAndID = util.formatNameAndID(group)
-    
+def addCanvasUsersToGroup(group, courseUsers, instructorLog):
+    """Add new users to a specific ArcGIS group"""
+
     logger.info("addCanvasUsersToGroup: enter")
+    groupNameAndID = util.formatNameAndID(group)
     
     if len(courseUsers) == 0:
         logger.info('No new users to add to ArcGIS Group {}'.format(groupNameAndID))
+        instructorLog += "No new users were added.\n\n"
         return instructorLog
 
     logger.info('Adding Canvas Users to ArcGIS Group {}: {}'.format(groupNameAndID, courseUsers))
     # ArcGIS usernames are U-M uniqnames with the ArcGIS organization name appended.
     arcGISFormatUsers = formatUsersNamesForArcGIS(courseUsers)
     logger.debug("addCanvasUsersToGroup: formatted: {}".format(arcGISFormatUsers))
-    
-    results = group.add_users(arcGISFormatUsers)
-    logger.debug("adding: results: {}".format(results))
 
-    usersNotAdded = results.get('notAdded')
-    """:type usersNotAdded: list"""
     usersCount = len(arcGISFormatUsers)
-    usersCount -= len(usersNotAdded) if usersNotAdded else 0
+    listsOfFormattedUsernames = util.splitListIntoSublists(arcGISFormatUsers, 20)
+    usersNotAdded = []
+
+    for listOfFormattedUsernames in listsOfFormattedUsernames:
+        try:
+            results = group.add_users(listOfFormattedUsernames)
+            logger.debug("adding: results: {}".format(results))
+            usersNotAdded += results.get('notAdded')
+        except RuntimeError as exception:
+            logger.error('Exception while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            return None
+
+    usersCount -= len(usersNotAdded)
     logger.debug("usersCount: {}".format(usersCount))
-    logger.debug("aCUTG: instructorLog 1: [{}]".format(instructorLog))
```

Comment:
Are these debug log lines really of no value?

Reply:
I decided to remove these for a few reasons. First, there was no equivalent in the removal function. Second, I was confused by what `aCUTG` meant, and it feels like that would be confusing for others. Third, if we are accumulating an `instructorLog` string here, why do we need to log this multiple times? Because it seems unlikely that we're going to be experiencing errors while concatenating strings, I opted for simply relying on the `logger.debug` message with the entire log at the end of each function. Does that make sense?